doc/source/coding-style/formatting-tools.rst

```diff
@@ -0,0 +1,241 @@
+   force_sort_within_sections = true
+   line_length = "<length>"
+   default_section = "THIRDPARTY"
+   src_paths = ["doc", "src", "tests"]
+
+
+Flake8
+------
+The goal of `flake8` is to act as a `PEP 8`_ compliance checker. Again, it is
+important to make sure that if this tool is being used with `black`_, no
+conflicts arise.
+
+The following configuration is the minimum one to setup `flake8`_ together with
+`black`_ one.
+
+The configuration for `flake8`_ must be specified in a ``.flake8`` file.
+
+.. code-block:: toml
+
+   [flake8]
+   max-line-length = 88
+   extend-ignore = E203
+
+
+Pre-commit
+----------
+To make sure that every commit you made is compliant with the code style
+guidelines of PyAnsys, you can take advantage of `pre-commit`_ in your project.
+Every time you stage some changes and once you commit those, `pre-commit`_ will
+only allow you to do so if all the defined hooks succeedd.
+
+The configuration for `pre-commit`_ must be defined in a
+`.pre-commit-config.yaml` file. The following lines present a minimum
+`pre-commit`_ configuration which includes both code and documentation
+formatting tools.
+
+
+.. code-block:: yaml
+
+    repos:
+    
+    - repo: https://github.com/psf/black
+      rev: X.Y.Z
+      hooks:
+      - id: black
+    
+    - repo: https://github.com/pycqa/isort
+      rev: X.Y.Z
+      hooks:
+      - id: isort
+    
+    - repo: https://github.com/PyCQA/flake8
+      rev: X.Y.Z
+      hooks:
+      - id: flake8
+    
+    - repo: https://github.com/codespell-project/codespell
+      rev: vX.Y.Z
+      hooks:
+      - id: codespell
+    
+    - repo: https://github.com/pycqa/pydocstyle
+      rev: X.Y.Z
+      hooks:
+      - id: pydocstyle
+        additional_dependencies: [toml]
+        exclude: "tests/"
+
+Installing ``pre-commit``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+You can install ``pre-commit`` by running:
+
+.. code-block:: bash
+
+    python -m pip install pre-commit
+
+Then, make sure you install it as a ``Git hook`` by running:
+
+.. code-block:: bash
+
+    pre-commit install
+
+Using ``pre-commit``
+~~~~~~~~~~~~~~~~~~~~
+From then on, pre-commit will automatically trigger every time you try to commit
+a change. If any of the hooks defined in `.pre-commit-config.yaml` fails, you
+will need to fix the failing files, stage the new changes and try to commit
+those again.
+
+If you want to manually run ``pre-commit``, you can execute:
+
+.. code-block:: bash
+
+    pre-commit run --all-files --show-diff-on-failure
+
+Previous command will show the current and expected style of the code if any of
+the hooks fail.
+
+
+Using ``pre-commit``
+~~~~~~~~~~~~~~~~~~~~
+
+Tox
+---
+A tool you may consider to use in your project is `tox`_. This tool is an
+automation tool similar to `Make`_ but with the advantage of allowing to test
+your package in a temporary virtual environment. This guarantees reproducible
+builds, as your package is no longer tested in "local" mode but in isolated
+form.
```

Comment:
```suggestion
A tool you might consider using in your project is `tox`_. While this automation
tool is similar to `Make`_, it supports testing of your package in a temporary
virtual environment. Being able to test your package in isolation rather than in
"local"  mode guarantees reproducible builds.
```